visualizer/Core/Audio/Views/SoundView.swift

```diff
@@ -108,10 +117,19 @@ struct SoundView: View {
             // show the DisplayDrawerView on DisplayDrawerButton (the one with layers icon) click
             ZStack {
                 DisplayDrawerView(isShowing: $showDisplayDrawer,
-                                 isShowingModal: $isShowingModal
+                                 isShowingModal: $isShowingDisplayDrawerModal
                 ).environmentObject(vm.DisplayDrawerVM)
                 
             }
+            ZStack{
+                RecordingAnalyticsDrawerView(
+                    isShowing: $showRecordingAnalyticsDrawer,
+                    isShowingModal: $isShowingRecordingAnalyticsDrawerModal,
+                    onClose: {
+                        showRecordingAnalyticsDrawer.toggle()
+                        isShowingRecordingAnalyticsDrawerModal.toggle()
+                })
```

Comment:
App crash on SoundView on physical device as you forget to pass the environmentObject here. You did it on the preview so that it can render on canvas

Add this line to solve:
```swift
 .environmentObject(vm.RecordingAnalyticsVM)
```